src/GitHub.VisualStudio/Base/EnsureLoggedInSection.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Reactive.Linq;
+using GitHub.Api;
+using GitHub.Extensions;
+using GitHub.Models;
+using GitHub.Services;
+using GitHub.UI;
+using GitHub.VisualStudio.Base;
+using Microsoft.TeamFoundation.MVVM;
+using System.Globalization;
+using GitHub.Primitives;
+
+namespace GitHub.VisualStudio.TeamExplorer.Sync
+{
+    public class EnsureLoggedInSection : TeamExplorerSectionBase
+    {
+        readonly IRepositoryHosts hosts;
+        readonly IVSServices vsServices;
+
+        public EnsureLoggedInSection(ISimpleApiClientFactory apiFactory, ITeamExplorerServiceHolder holder,
+            IConnectionManager cm, IRepositoryHosts hosts, IVSServices vsServices)
+            : base(apiFactory, holder, cm)
+        {
+            IsVisible = false;
+            this.hosts = hosts;
+            this.vsServices = vsServices;
+        }
+
+        public override void Initialize(IServiceProvider serviceProvider)
+        {
+            base.Initialize(serviceProvider);
+            CheckLogin();
+        }
+
+        protected override void RepoChanged()
+        {
+            base.RepoChanged();
+            CheckLogin();
+        }
+
+        async void CheckLogin()
```

Comment:
Unless this is used as an event handler delegate (which it doesn't appear to be), we should [avoid using `async void`](http://haacked.com/archive/2014/11/11/async-void-methods/) and return a `Task` instead.